packages/blade/src/components/Amount/Amount.tsx

```diff
@@ -0,0 +1,193 @@
+import type { ReactElement } from 'react';
+
+import {
+  currencyAbbreviationsMapping,
+  currencylocaleMapping,
+  currencyPrefixMapping,
+} from './amountTokens';
+import AmountValue, { getAffixFontSize } from './AmountValue';
+import { StyledAmount } from './StyledAmount';
+import { BaseText } from '~components/Typography/BaseText';
+import type { Feedback } from '~tokens/theme/theme';
+import type { BaseTextProps } from '~components/Typography/BaseText/types';
+import BaseBox from '~components/Box/BaseBox';
+import type { TestID } from '~src/_helpers/types';
+import { metaAttribute, MetaConstants } from '~utils';
+
+export const suffixTypes = {
+  HUMANIZE: 'humanize',
+  DECIMALS: 'decimals',
+  NONE: 'none',
+};
+
+type currencyType = 'INR' | 'MYR';
+
+type AmountProps = {
+  /**
+   * The value to be rendered within the component.
+   *
+   */
+  value: number;
+  /**
+   * Sets the intent of the amount.
+   *
+   * @default 'neutral'
+   */
+  intent?: Feedback | undefined;
+  /**
+   * Sets the variant of the amount.
+   *
+   * @default 'heading'
+   */
+  size?:
+    | `title-medium`
+    | `title-small`
+    | `heading-large`
+    | `heading-large-bold`
+    | `heading-small`
+    | `heading-small-bold`
+    | `body-medium`
+    | `body-medium-bold`
+    | `body-small`
+    | `body-small-bold`;
```

Comment:
just thought a bit more about this. If we add new sizes in title or text, it will automatically update the acceptable size on Amount but we actually won’t have that support on Amount and that would not render as a consumer would expect. Rather its better if we add a new size to Title and then get an issue raised that Amount doesn’t support that size.
Seems its safer to keep literal strings for size.